src/Components/WebAssembly/Build/src/targets/BuiltInBclLinkerDescriptor.xml

```diff
@@ -24,4 +24,15 @@
     <type fullname="WebAssembly.Net.Http.HttpClient.FetchCredentialsOption" />
     <type fullname="WebAssembly.Net.Http.HttpClient.WasmHttpMessageHandler" />
   </assembly>
+
+  <assembly fullname="System.Text.Json">
+    <!-- S.T.J. uses Activator.CreateInstance to instantiate converters, so we need to preserve default constructors.
+    For safety, do this for all converter types, even though most of them are preserved anyway due to being referenced
+    statically. It's only JsonStringEnumConverter that currently isn't referenced statically.
+    The following is a workaround for https://github.com/dotnet/aspnetcore/issues/19086.
+    The underlying issue is reported in the runtime repo at https://github.com/dotnet/runtime/issues/34449 -->
+    <type fullname="System.Text.Json.Serialization.*Converter">
```

Comment:
Ahh, I see. The embedded linker config includes `JsonConverterEnum<T>`, but the not the string one.

Reply:
Is there an embedded linker config in S.T.J? I didn't spot one.

AFAIK the way the other converters are avoiding having their constructors get culled is that they are referenced statically from some bits of code that cache instances.

Reply:
Yup, here's what it says:

```xml
<linker>
  <assembly fullname="System.Text.Json">
    <type fullname="System.Text.Json.JsonPropertyInfoNullable`2">
      <!-- Instantiated via reflection -->
      <method name=".ctor" />
    </type>
    <type fullname="System.Text.Json.Serialization.JsonPropertyInfoNotNullableContravariant`4">
      <!-- Instantiated via reflection -->
      <method name=".ctor" />
    </type>
    <type fullname="System.Text.Json.Serialization.Converters.JsonConverterEnum`1">
      <!-- Instantiated via reflection -->
      <method name=".ctor" />
    </type>
    <type fullname="System.Text.Json.Serialization.Converters.JsonKeyValuePairConverter`2">
      <!-- Instantiated via reflection -->
      <method name=".ctor" />
    </type>
  </assembly>
</linker>
```